### PR TITLE
Libedit

### LINKED ISSUE #4209: Package libedit
This is so that we can have a libedit pinned to ncurses 5.9 which is the conda-forge pin for now

See https://github.com/conda-forge/octave-feedstock/issues/6 and https://github.com/bioconda/bioconda-recipes/issues/6219

Anaconda recipe is here, https://github.com/AnacondaRecipes/aggregate/tree/master/libedit-feedstock/recipe

cc @jakirkham, @msarahan, @mingwandroid

### PR DESCRIPTION
**Pull request by scopatz**
Closes #4209  CC @isuruf @msarahan @jakirkham

### CODE REVIEW

**Comment by scopatz** [comment]
Anyone in @conda-forge/staged-recipes know why this this trying to be built three times on linux?

**Comment by scopatz** [comment]
All 3 cases seem to be exactly the same...

File: recipes/libedit/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - toolchain
+    - automake
+    - autoconf
+    - m4
+    - perl
```
Comment (isuruf) [inline recipes/libedit/meta.yaml]: You need to pin perl

Comment (scopatz) [reply recipes/libedit/meta.yaml]: Ok, thanks, to what though?  https://github.com/conda-forge/conda-forge.github.io/blob/master/scripts/pin_the_slow_way.py doesn't list a pinning

Comment (isuruf) [reply recipes/libedit/meta.yaml]: `5.22.0.1`. see https://github.com/conda-forge/conda-smithy/issues/415

Comment (scopatz) [reply recipes/libedit/meta.yaml]: Ok! Thanks!

Comment (scopatz) [reply recipes/libedit/meta.yaml]: That worked!

**Comment by jakirkham** [comment]
Thanks for tackling this one @scopatz. :)

**Comment by scopatz** [comment]
No problem!

File: recipes/libedit/0001-Also-look-in-libtinfo-for-tgetent.patch [hunk 2]
```diff
@@ -0,0 +1,100 @@
+diff -urN work.old/configure work/configure
+--- work.old/configure	2017-08-22 14:16:00.621910386 +0100
++++ work/configure	2017-08-22 14:17:20.482361609 +0100
+@@ -586,7 +586,7 @@
+ 
+ # Identity of this package.
+ PACKAGE_NAME='libedit'
+-PACKAGE_TARNAME='libedit-20170329'
++PACKAGE_TARNAME='libedit-20170822'
```
Comment (isuruf) [inline recipes/libedit/0001-Also-look-in-libtinfo-for-tgetent.patch]: Why this change?

Comment (scopatz) [reply recipes/libedit/0001-Also-look-in-libtinfo-for-tgetent.patch]: I think this is just to fool the package.  I picked up this patch from another anaconda libedit package.

**Comment by jakirkham** [comment]
Yeah if they hard coded the dates and they don't match, we have to patch them if they cause problems when wrong.

LGTM. If there are no more comments, would be fine merging it. Did anyone try this out on macOS? Just thinking we can get faster feedback than waiting for Travis CI.

**Comment by isuruf** [comment]
Works for me https://travis-ci.org/isuruf/staged-recipes/builds/293889614?utm_source=email&utm_medium=notification

**Comment by jakirkham** [comment]
Awesome! Thanks for checking @isuruf. Also thanks again @scopatz for the submission.
